### ISSUE TITLE
Visit void sprite planets : Gone after update

### ISSUE THREAD

**Issue by StickyDigit** [endless-sky/endless-sky#7632]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

After update, using my existing save I get this message:
> These active missions or jobs were deactivated due to a missing definition - perhaps you recently removed a plugin?
> "Visit void sprite planets"
> They will be reactivated when the necessary plugin is reinstalled.

..and of course that mission is gone.

### Steps to Reproduce

I had previously installed 0.9.8 via "apt install endless-sky" on Linuxmint.

Played for ages. Had recently taken the mission to land on the gas giant.

To upgrade, I did "apt autoremove --purge endless-sky" then built 0.9.16.1 from source. The same issue happens using the appimage.

No plugins added at any stage.

### Expected Behavior

I would expect the upgraded version to not break saved games.

### Link to save file

[Sticky Digit.txt](https://github.com/endless-sky/endless-sky/files/10031749/Sticky.Digit.txt)

### Operating System

Linuxmint 21

### Game Source

Other (please describe)

### Game Version

v0.9.16.1

### Additional Information

I still have the Gascraft I had bought. It's parked at Arculus FWIW.

Parsing the config file for errors:
```
$endless-sky -p
Warning: event "remnant: gascraft" is referred to, but not fully defined.
Warning: government " Drak " is referred to, but not fully defined.
Warning: government " Kor Mereti " is referred to, but not fully defined.
Warning: ship "Gascraft" is referred to, but not fully defined.
Warning: shipyard "Remnant Gascraft" is referred to, but has no ships.
Warning: system "Membulem" is referred to, but not fully defined.
Parse completed.
```

**Comment by StickyDigit** [comment]
Yes that works. Thank you.

Could/Should that fix be built-in, in order to auto-update legacy saves?

**Comment by warp-core** [comment]
It could, and I think it should, I'll work on that shortly. So, leave this open for now.

**Comment by ThrawnCA** [comment]
This is particularly relevant because the mission in question is difficult to complete, and could easily be placed in the "too hard, come back later" basket.

And doubly so because of the bug that causes you to instantly fail all your active delivery missions if you park your escorts outside the system, land on a gas giant (auto-saving), and then load your saved game - which could easily happen if the Drak blow you up. So you can't take several bites at it, unless you first go away and clear your job queue or are willing to fail them all.

So there may be more players with this mission pending than would otherwise be the case.

**Comment by warp-core** [comment]
> And doubly so because of the bug that causes you to instantly fail all your active delivery missions if you park your escorts outside the system, land on a gas giant (auto-saving), and then load your saved game - which could easily happen if the Drak blow you up. So you can't take several bites at it, unless you first go away and clear your job queue or are willing to fail them all.

That one is actually fixed (in continuous) now!
a208608f63d90792b9f1d0b5fac6478a3fdba3bc

**Comment by ThrawnCA** [comment]
> > And doubly so because of the bug that causes you to instantly fail all your active delivery missions if you park your escorts outside the system, land on a gas giant (auto-saving), and then load your saved game - which could easily happen if the Drak blow you up. So you can't take several bites at it, unless you first go away and clear your job queue or are willing to fail them all.
> 
> That one is actually fixed (in continuous) now! [a208608](https://github.com/endless-sky/endless-sky/commit/a208608f63d90792b9f1d0b5fac6478a3fdba3bc)

Yep, but this issue is specifically about players on 0.9.8... And it's a barrier to updating.

**Comment by Amazinite** [comment]
Fixed by https://github.com/endless-sky/endless-sky/pull/7633.
